includes/Classifai/Providers/OpenAI/ChatGPT.php

```diff
@@ -61,6 +61,43 @@ public function __construct( $service ) {
 		);
 	}
 
+	/**
+	 * Determine if the current user can access the feature
+	 *
+	 * @param {string} $feature       Feature to check.
+	 * @param {array}  $feature_roles Roles that have access to the feature.
+	 *
+	 * @return bool
+	 */
+	public function user_can_access( $feature, $feature_roles = [] ) {
+		$access   = false;
+		$settings = $this->get_settings();
+
+		if (
+			! empty( $settings ) &&
+			isset( $settings['authenticated'] ) &&
+			false !== $settings['authenticated'] &&
+			isset( $settings[ $feature ] ) &&
+			'no' !== $settings[ $feature ]
```

Comment:
Similar to the above, not sure I'd put this check here, was this checks if a feature is enabled, not if the user has access to the feature. I think I'd recommend removing this and instead adding our role checking in this method (which currently that hasn't been moved). So if we aren't authenticated or if the feature isn't turned on, no one will have access. But if both of those are true, we can then run this method to determine if an individual user should have access (by default, just checking if their role is allowed but developers can hook in and change access on a per user basis if needed).

Reply:
@dkotter, The fix for this is now ready for review. 

cc @jeffpaul